src/couch/src/couch_db_plugin.erl

```diff
@@ -34,11 +34,13 @@
 validate_dbname(DbName, Normalized, Default) ->
     maybe_handle(validate_dbname, [DbName, Normalized], Default).
 
-before_doc_update(Db, Doc0) ->
+before_doc_update(Db, Doc0, UpdateType) ->
     Fun = couch_db:get_before_doc_update_fun(Db),
-    case with_pipe(before_doc_update, [Doc0, Db]) of
-        [Doc1, _Db] when is_function(Fun) -> Fun(Doc1, Db);
-        [Doc1, _Db] -> Doc1
+    case with_pipe(before_doc_update, [Doc0, Db, UpdateType]) of
+        [Doc1, _Db, _UpdateType] when is_function(Fun) ->
+            Fun(Doc1, Db, _UpdateType);
```

Comment:
Ahh. I didn't notice earlier that you are passing `_UpdateType` to Fun. In this case it is better to rename it into `UpdateType1` or something. According to erlang's conventions (roughly).

- Use new name for variable when the value need to be changed (usually the integer is added as suffix)
- Use `_Name` style for variables you don't care about.

In this case we do care about the variable since we are passing it to another function. Also we cannot use `UpdateType` since the value returned from `with_pipe` can differ from the passed value. In this case we would have a badmatch.